tests/src/Unit/Plugin/Notifier/SmsTest.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+namespace Drupal\Tests\message_notify\Unit\Plugin\Notifier;
+
+use Drupal\Component\Uuid\UuidInterface;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Language\Language;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\Core\Render\RendererInterface;
+use Drupal\message\MessageInterface;
+use Drupal\message_notify\Plugin\Notifier\Sms;
+use Drupal\sms\Exception\NoPhoneNumberException;
+use Drupal\sms\Message\SmsMessageResultStatus;
+use Drupal\sms\Provider\PhoneNumberProviderInterface;
+use Drupal\Tests\UnitTestCase;
+use Drupal\user\UserInterface;
+use Exception;
+use Prophecy\Argument;
+
+/**
+ * Unit tests for the SMS notifier.
+ *
+ * @coversDefaultClass \Drupal\message_notify\Plugin\Notifier\Sms
+ *
+ * @group message_notify
+ */
+class SmsTest extends UnitTestCase {
+
+  /**
+   * Mocked configuration.
+   *
+   * @var array
+   */
+  protected $configuration;
+
+  /**
+   * Mocked plugin id.
+   *
+   * @var string
+   */
+  protected $pluginId;
+
+  /**
+   * Mocked plugin definition.
+   *
+   * @var array
+   */
+  protected $pluginDefinition;
+
+  /**
+   * Mocked logger.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * Mocked entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Mocked renderer.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * Mocked phone number provider.
+   *
+   * @var \Drupal\sms\Provider\PhoneNumberProviderInterface
+   */
+  protected $phoneNumberProvider;
+
+  /**
+   * Test setup.
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // SmsMessage.php#L326 is calling uuid service directly. Need to setup
+    // fake container for this test.
+    $container = new ContainerBuilder();
+    $container->set('uuid', $this->getMockUuidService());
```

Comment:
You could subclass \Drupal\sms\Message\SmsMessage in your test file and override the uuid generator method

Reply:
This one I couldn't wrap my head around. I saw one test extending `SmsMessage` and was planning to do that. But not sure how that would work when the usage of `SmsMessage` is [in the plugin, not the test](https://github.com/Gizra/message_notify/pull/74/files#diff-429ad833ba1e7c88431c0511454fc333R90).

I'm not yet that familiar with Drupal's injection system 😓 Any pointers would help.